src/lib.rs

```diff
@@ -136,10 +136,12 @@ impl Piranha {
   fn perform_cleanup(&mut self) {
     // Setup the parser for the specific language
     let mut parser = Parser::new();
+    let piranha_args = self.rule_store.piranha_args().clone();
```

Comment:
Question: What changed that requires `.clone()` here? The only place I see any reference within this object (not sure if the correct Rust terminology) being passed somewhere new is with `current_global_substitutions`, but we are calling `.clone()` there at the level of `piranha_args.input_substitutions()`. What am I missing?

Reply:
Aah! I had to clone it because I extracted `piranha_args` as a variable. 

`self.rule_store.piranha_args()` returns `&PiranhaArguments`.  So if we don't clone it here, it is going to be a "Immutable borrow" for `self.rule_store`. Instead of borrowing I just cloned it :| 
If I drop the `clone()` here, rust compiler complains that I am using `self.rule_store` as immutable here and mutable (later below in the function). 

The work-around for this is to "inline" this variable.